src/NuGet.Clients/PackageManagement.VisualStudio/Migrate/ProjectJsonToPackageRefMigrator.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using NuGet.Common;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.ProjectModel;
+using Task = System.Threading.Tasks.Task;
+using EnvDTE;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    public class ProjectJsonToPackageRefMigrator
+    {
+        private LegacyCSProjPackageReferenceProject _project;
+        private IVsBuildPropertyStorage _buildPropertyStorage;
+        private Project _envdteProject;
+        private IVsHierarchy _ivsHierarchy;
+        public ProjectJsonToPackageRefMigrator(LegacyCSProjPackageReferenceProject project, Project envDteProject, IVsHierarchy hierarchy)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+            _project = project;
+            _envdteProject = envDteProject;
+            _ivsHierarchy = hierarchy;
+            _buildPropertyStorage = hierarchy as IVsBuildPropertyStorage;
```

Comment:
If that's required, change the constructor argument type to `IVsBuildPropertyStorage`.